osv/models.py

```diff
@@ -905,6 +934,16 @@ class AliasDenyListEntry(ndb.Model):
   bug_id: str = ndb.StringProperty()
 
 
+class UpstreamGroup(ndb.Model):
+  """Upstream group for storing transitive upstreams of a Bug"""
```

Comment:
Can you expand the description here to explain why this is in a separate table rather than directly on the bug entry itself? 

I.e. to prevent additional race conditions, by having it in a separate table only worker will modify Bug's directly.